### ISSUE TITLE
apply_patches CMakeFunction shouldn't checkout existing branch

### ISSUE THREAD

**Issue by PawelJurek** [intel/opencl-clang#215]
Function "apply_patches" checks out the branch, where patches are applied if it already exists: 
https://github.com/intel/opencl-clang/blob/51fd5b94830ebdc2c618da24217d08a27de9ca36/cmake/modules/CMakeFunctions.cmake#L71

I think the logic here might lead to unexpected issues, e.g. if you download new opencl-clang commit, where a new patch is introduced, it won't be applied if the branch is not manually deleted.

Maybe the function should delete the branch at first step?

**Comment by hewj03** [comment]
yes I think the logic needs to be refined so that new patch will be applied.

**Comment by haonanya** [comment]
Besides, function that checks if the the patch is present in current local branch https://github.com/intel/opencl-clang/blob/ocl-open-100/cmake/modules/CMakeFunctions.cmake#L52 also doesn't work. It seems that commit id from one patch generated by git format-patch is always not same as the result from git am, see example from https://stackoverflow.com/questions/8550400/should-git-apply-or-git-am-come-up-with-the-same-hash.
